src/main/java/skylark/command/Command.java

```diff
@@ -0,0 +1,329 @@
+package skylark.command;
+
+import skylark.skylark.SkylarkException;
+import skylark.task.Deadline;
+import skylark.task.Event;
+import skylark.task.Task;
+import skylark.task.TaskList;
+import skylark.task.ToDo;
+
+/**
+ * Represents a command based on user input. <br><br>
+ * A <code>Command</code> object corresponds to
+ * a particular command selected by the user e.g., user selects <code>list</code>.
+ */
+public abstract class Command {
+
+    /** String representing the input keyed by the user. */
+    private final String input;
+
+    private Command(String input) {
+        this.input = input;
+    }
+
+    /**
+     * Interprets the command based on keywords.
+     * If there are no matching commands, a UnknownCommand is returned.
+     *
+     * @param command Input string from the user.
+     * @return A Command object based on interpreted input.
+     */
+    public static Command createCommand(String command) {
+        if (command.equals(CommandList.COMMAND_LIST.toString())) {
+            return new ListCommand(command);
+        } else if (command.length() >= 4 && command.startsWith(CommandList.COMMAND_DONE.toString())) {
+            return new DoneCommand(command);
+        } else if (command.length() >= 6 && command.startsWith(CommandList.COMMAND_UNDONE.toString())) {
+            return new UndoneCommand(command);
+        } else if (command.length() >= 4 && command.startsWith(CommandList.COMMAND_TODO.toString())) {
+            return new TodoCommand(command);
+        } else if (command.length() >= 8 && command.startsWith(CommandList.COMMAND_DEADLINE.toString())) {
+            return new DeadlineCommand(command);
+        } else if (command.length() >= 5 && command.startsWith(CommandList.COMMAND_EVENT.toString())) {
+            return new EventCommand(command);
+        } else if (command.length() >= 6 && command.startsWith(CommandList.COMMAND_DELETE.toString())) {
+            return new DeleteCommand(command);
+        } else if (command.length() >= 5 && command.startsWith(CommandList.COMMAND_FIND.toString())) {
+            return new FindCommand(command);
```

Comment:
I like the way you checked each case thoroughly using .length() and .startsWith(). A possible alternative could be to read the first element in a String array after using .split(" ")